### ISSUE TITLE
VSCode launch/build target fails

### ISSUE THREAD

**Issue by seemsindie** [floooh/fips#283]
Hello,

I think I found a bug but I am not sure, it might as well be me. I have the `fips-workspace` folder containing my fips projects and I generated a fresh project to learn sokol, and as soon as I added `sokol_shader(1-triangle.glsl ${slang})` to one of the targets I got the following error building from VSCode.

```
[main] Building folder: playground-sokol 1-2-1-triangle
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/timelord/Projects/fips-workspace/playground-sokol/build --config Debug --target 1-2-1-triangle -j 18 --
[build] [1/3  33% :: 0.047] cd /home/timelord/Projects/fips-workspace/playground-sokol && /usr/bin/python3.10 /home/timelord/Projects/fips-workspace/playground-sokol/build/fips-gen.py /home/timelord/Projects/fips-workspace/playground-sokol/build/fips_codegen.yml
[build] FAILED: src/1-2-hello-triangle/CMakeFiles/ALL_GENERATE /home/timelord/Projects/fips-workspace/playground-sokol/build/src/1-2-hello-triangle/CMakeFiles/ALL_GENERATE 
[build] cd /home/timelord/Projects/fips-workspace/playground-sokol && /usr/bin/python3.10 /home/timelord/Projects/fips-workspace/playground-sokol/build/fips-gen.py /home/timelord/Projects/fips-workspace/playground-sokol/build/fips_codegen.yml
[build] /usr/bin/python3.10: can't open file '/home/timelord/Projects/fips-workspace/playground-sokol/build/fips-gen.py': [Errno 2] No such file or directory
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

Reading the error it appears it's looking for `fips-gen.py` in the wrong directory (looks like cwd issue maybe?), it's trying to read from the build directory from the root of the project (that folder is auto-generated by vscode) instead of the project folder inside of the `fips-build` directory.

I ran `./fips gen` and then `./fips open` to use the project properly, and IntelliSense works great (which I need more than running the project) but when I try to build or run it fails with that error.

When running from the terminal both `./fips build` and `./fips run 1-2-1-triangle` run successfully, it builds and runs it without any issue. The first example `./fips run 1-2-1-window` or from vscode run as well. Debugging projects works too.

I uploaded the code at https://github.com/seemsindie/sokol-playground

**Comment by floooh** [comment]
I'll take a look...

**Comment by floooh** [comment]
Hmm I can't reproduce the error on macOS (I just had to fix all occurances of "python" with "python3", because the recent update to macOS 12.3 removes the python executable).

Here's what I did:

```
mkdir fips-workspace
cd fips-workspace
git clone https://github.com/seemsindie/sokol-playground
cd sokol-playground
# (before the next command I had to change the python shebang in 'fips' to python3)
./fips set config osx-vscode-debug
./fips gen
./fips open
[VSCode opens]
[in VSCode: Cmd+Shift+B to build the project]
[in VSCode terminal: ./fips run 1-2-1-triangle
```

...which then produces:
<img width="834" alt="Screenshot 2022-03-19 at 17 17 01" src="https://user-images.githubusercontent.com/1699414/159129750-cb269b23-14b9-4d27-b7f6-1434f52d93aa.png">

Can you provide more details what you did (also, what OS are you on?)

The only thing that looks unusual to me is the targets starting with a number (e.g. ```1-2-1-triangle``` instead of smth like ```triangle-1-2-1```, maybe that triggers some obscure problem on some other platforms).

**Comment by seemsindie** [comment]
I just took another look and I figured out that it's not really an issue with VSCode but it's an extension. It's just me being an idiot, sorry. I downloaded the insider's version so I can have a quick clean look at this. And immediately I noticed that I am missing the bottom bar (and that is what I mostly use for running/debugging, release/debug, changing kits, targets...).

![image](https://user-images.githubusercontent.com/1268576/159453066-0eb09b08-92b4-4505-9344-0f305e6fc300.png)

Now the insider's version I installed offered me to install the C/C++ extension pack which I did and got my bottom bar back :) So I am going to close this issue since `fips` work fine with VSCode. The issue is more with the CMake Tools extension from the pack.
